include/circt/Support/InstanceGraphInterface.td

```diff
@@ -0,0 +1,67 @@
+//===- InstanceGraphInterface.td - Interface for instance graphs --------*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file contains interfaces and other utilities for interacting with the
+// generic CIRCT instance graph.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_SUPPORT_INSTANCEGRAPH_INSTANCEGRAPHINTERFACE_TD
+#define CIRCT_SUPPORT_INSTANCEGRAPH_INSTANCEGRAPHINTERFACE_TD
+
+include "mlir/IR/AttrTypeBase.td"
+include "mlir/IR/OpBase.td"
+
+def InstanceGraphInstanceOpInterface : OpInterface<"InstanceOpInterface"> {
+  let description = [{
+    This interface provides hooks for an instance-like operation.
+  }];
+  let cppNamespace = "::circt::igraph";
+
+  let methods = [
+    InterfaceMethod<"Get the name of the instance",
```

Comment:
Must all instances have names?  This doesn't seem obviously true for all dialects.  It seems primarily a requirement for pretty graph printing.